### ISSUE TITLE
Use of Webpack aliases and loaders?

### ISSUE THREAD

**Issue by markerikson** [zinserjan/mocha-webpack#4]
I've got an existing Webpack config that runs my whole test suite and dependencies through Webpack and Babel.  That means that my test files themselves are written in ES6, including use of "import".  I also have a number of aliases for folders inside my `src` folder, ie, `$REPO/src/components/SomeComponent.jsx` can be referred to as simply "components/SomeComponent".  

I'm trying to set up mocha-webpack, and I'm pointing it at a slightly cut-down version of my existing Webpack test config (removed "entry" and "output", etc).  At the moment, it does not appear that it actually makes any use of my config's loaders or aliases.  Pointing it at a specific test file with ES6 syntax produced "Unexpected token" errors indicating unsupported syntax, and when I switched my "import" statements to "require" statements, it then complained that it couldn't find the files referred to by my aliases.

Is there a particular bit of configuration I need to pass to mocha-webpack to have it use the loaders on the test code, is there something obvious I'm missing, or does mocha-webpack not support transpiling the test files themselves?

**Comment by zinserjan** [comment]
> Is there a particular bit of configuration I need to pass to mocha-webpack to have it use the loaders on the test code, is there something obvious I'm missing

Just provide the path to your webpack config to mocha-webpack as argument. For example, if your webpack config is located at the project root:
`mocha-webpack --webpack-config webpack.config-test.js "src/**/*.test.js"`

> or does mocha-webpack not support transpiling the test files themselves?

No, the transpiling and bundling step before running tests is the main task of this project.

If you already tried the config argument, can you provide a repo with this failure?

**Comment by markerikson** [comment]
I've got a stripped-down version of my current prototype project published over at https://github.com/markerikson/react-redux-cesium-testing-demo.  Made a number of additional changes to my actual project in the last few days, but that sample repo is still mostly how I have things set up.

**Comment by markerikson** [comment]
Hmm.  I _may_ have actually gotten this working with my current project.  Among other things, was seeing errors like "SyntaxError: Unexpected token /", but it turns out that was apparently due to use of a devtool sourcemap setting that included "eval".  Removing that seems to let it work.  

For what it's worth, the full-blown command line here on my Windows box is:
`node_modules\.bin\cross-env NODE_ENV=test node_modules\.bin\mocha-webpack.cmd --webpack-config .\webpack\webpack.test.config.js --opts "test/mocha-webpack.opts" --watch --recursive --glob "*.spec.js" test`

I'll play around with this some more tomorrow, but this looks encouraging.

**Comment by markerikson** [comment]
Seems like it's working for me at this point.  I really was able to feed an existing Webpack config to it rather than having to create a modified version.  Biggest thing was needing to specify a devtool value that did not include "eval", as some piece in the pipeline seemed to choke on that.

I will say that while it consistently reruns my tests immediately if I edit a test, it seems to have a bit of issue rerunning things if I update a source file that a test is importing (like, say, adding a console.log statement to the actual code being tested).  Couple times I had to keep making tiny edits to the code repeatedly before something kicked in and it would actually notice that the actual source file had been changed and go re-run the tests.

Still, looks pretty sweet.  I'll have to get my sample project updated with this sometime soon.  Thanks!

**Comment by zinserjan** [comment]
> Seems like it's working for me at this point

Sounds good!

> Biggest thing was needing to specify a devtool value that did not include "eval", as some piece in the pipeline seemed to choke on that.

Thanks for the info. In my current project, devtool is just " #source-map",  but I can't remember why. Will try the other options when I find some time.

> Couple times I had to keep making tiny edits to the code repeatedly before something kicked in and it would actually notice that the actual source file had been changed and go re-run the tests.

Had this problem also with all kind of file watchers. It turned out that my IDE has a "safe write" mode. Since I disabled this option everything works :)

safe write mode:
>  ... a changed file is first saved in a temporary file. If the save operation succeeds, the file being saved is replaced with the saved file. (Technically, the original file is deleted and the temporary file is renamed.)
see https://www.jetbrains.com/help/idea/2016.1/system-settings.html

**Comment by markerikson** [comment]
Thanks for the pointer on the JetBrains "safe write" setting.  Was definitely seeing problems with app source edits not causing tests to re-run.  Just tried turning that setting off, and now those are working correctly as well.

Last couple days have been great.  This is actually my first real experience writing actual unit tests (a sad deficiency in my experience up until now).  With mocha-webpack in place, I've been able to rapidly iterate on writing tests, both in terms of adding new tests, learning the ins and outs of Chai+Sinon, and debugging stupid mistakes I've made in writing them :)

This library gets a total :+1: from me.  Thanks for writing it!
